internal/dao/stubs/pubkey_dao.go

```diff
@@ -28,39 +28,54 @@ func getPubkeyDao(ctx context.Context) (dao.PubkeyDao, error) {
 	return getPubkeyDaoStub(ctx)
 }
 
-func (stub *pubkeyDaoStub) Create(ctx context.Context, pk *models.Pubkey) error {
-	pk.ID = stub.lastId + 1
-	stub.store = append(stub.store, pk)
+func (stub *pubkeyDaoStub) Create(ctx context.Context, pubkey *models.Pubkey) error {
+	if pubkey.AccountID == 0 {
+		pubkey.AccountID = ctxAccountId(ctx)
+	}
+	if pubkey.AccountID != ctxAccountId(ctx) {
+		return dao.WrongTenantError
+	}
+
+	pubkey.ID = stub.lastId + 1
+	stub.store = append(stub.store, pubkey)
 	stub.lastId++
 	return nil
 }
 
-func (stub *pubkeyDaoStub) Update(ctx context.Context, pk *models.Pubkey) error {
+func (stub *pubkeyDaoStub) Update(ctx context.Context, pubkey *models.Pubkey) error {
+	if pubkey.AccountID == 0 {
+		pubkey.AccountID = ctxAccountId(ctx)
+	}
+	if pubkey.AccountID != ctxAccountId(ctx) {
+		return dao.WrongTenantError
+	}
+
 	for idx, p := range stub.store {
-		if p.ID == pk.ID {
-			stub.store[idx] = pk
+		if p.ID == pubkey.ID {
+			stub.store[idx] = pubkey
 			return nil
 		}
 	}
 	return NewRecordNotFoundError(ctx, "Pubkey")
 }
 
 func (stub *pubkeyDaoStub) GetById(ctx context.Context, id int64) (*models.Pubkey, error) {
-	for _, acc := range stub.store {
-		if acc.ID == id {
-			return acc, nil
+	for _, pk := range stub.store {
+		if pk.AccountID == ctxAccountId(ctx) && pk.ID == id {
+			return pk, nil
 		}
 	}
 	return nil, NewRecordNotFoundError(ctx, "Pubkey")
 }
 
 func (stub *pubkeyDaoStub) List(ctx context.Context, limit, offset int64) ([]*models.Pubkey, error) {
+	// not tenant filtered, should it?
```

Comment:
I guess it should but to which account id do we compare?

Reply:
I'll add it, it is just me being lazy... lets not be lazy :stuck_out_tongue:

Reply:
Alright, done :+1: